### PR TITLE
Add Snowflake support for `assertions()` and `assertions_filter()`

### PR DESCRIPTION
**Pull request by b-per**
Add support for `assertions()` and `assertions_filter()` in Snowflake

Currently, the following was not tested/implemented yet:
- `generic_assertions` (not tested but it should work I think)
- `__unique__` helper
- `__not_null__` helper

I could add support for those (might take a bit), or we could update the README maybe to add a table showing what feature is supported by which adapter. I am happy to do so if need be.

In addition, if it's OK, I will add an issue to have some integration tests in the package so that people can more easily add new adapters 😉

### CODE REVIEW

File: macros/_assertions_expression.sql [hunk 1]
```diff
@@ -31,3 +31,33 @@ ARRAY_CONCAT(
...
+    IFF(
+        COALESCE({{ expression }}, {{ null_as_exception }}) = FALSE,
+        '{{ assertion_id }}',
+        CAST(NULL AS STRING)
+    ),
+    {%- endfor %}
+    CAST(NULL AS STRING)
+) AS {{ column }}
+
+{%- endmacro %}
```
Comment (AxelThevenot) [inline macros/_assertions_expression.sql:63]: Can you jus add a new line at the end of your file? 👍

File: macros/assertions_filter.sql [hunk 2]
```diff
@@ -64,3 +64,29 @@ EXISTS (
...
+{{- 'NOT ' if reverse else '' -}}
+{%- if include_list is not none -%}
+ARRAY_SIZE(ARRAY_INTERSECTION({{ column }},ARRAY_CONSTRUCT('{{ include_list | join("\', \'")}}'))) = 0
+{%- elif exclude_list is not none -%}
+ARRAY_SIZE(ARRAY_EXCEPT({{ column }},ARRAY_CONSTRUCT('{{ exclude_list | join("\', \'")}}'))) = 0
+{%- else -%}
+ARRAY_SIZE({{ column }}) = 0
+{%- endif -%}
+
+{%- endmacro %}
```
Comment (AxelThevenot) [inline macros/assertions_filter.sql:92]: Same here

File: macros/assertions_filter.sql [hunk 3]
```diff
@@ -64,3 +64,29 @@ EXISTS (
...
+        )
+    }}
+{%- endif -%}
+
+{#- Generate filtering expression  -#}
+{{- 'NOT ' if reverse else '' -}}
+{%- if include_list is not none -%}
+ARRAY_SIZE(ARRAY_INTERSECTION({{ column }},ARRAY_CONSTRUCT('{{ include_list | join("\', \'")}}'))) = 0
+{%- elif exclude_list is not none -%}
+ARRAY_SIZE(ARRAY_EXCEPT({{ column }},ARRAY_CONSTRUCT('{{ exclude_list | join("\', \'")}}'))) = 0
```
Comment (AxelThevenot) [inline macros/assertions_filter.sql:87]: Great job with intersection and union !

**Review by AxelThevenot — CHANGES_REQUESTED** [review]

**Comment by AxelThevenot** [comment]
Thank you for this code ! Really nice to see how it is implemented !

Just an easy change resquested but it will be merge then

Is it ok for you if I make a prelease of it instead of a release , 
You'll need to import the package with `git` not with the hub I think

**Comment by AxelThevenot** [comment]
Good idea the markdown table for supported things in the README
I'll do it :)

And great idea also for the unit tests!
If you have any sample on your side, do not hesitate to instantiates those tests

File: macros/assertions_filter.sql (hunk 3 above)
Comment (b-per) [reply macros/assertions_filter.sql:87]: This should also work with databricks (and maybe other adapters as well) I think

**Comment by b-per** [comment]
I created the issue about integration tests, with a bit more info, here: #24 

And yes, you can do a pre-release if you want. Those can actually also be installed from the Hub I believe.

**Review by AxelThevenot — APPROVED** [review]

**Comment by AxelThevenot** [comment]
Release 1.1.0a

**Pull request merged**